src/Models/NonhydrostaticModels/boundary_mass_fluxes.jl

```diff
@@ -0,0 +1,239 @@
+open_boundary_mass_flux(model, bc::ZIOBC, ::Val{:bottom}, w) = zero(model.grid)
+open_boundary_mass_flux(model, bc::ZIOBC, ::Val{:top}, w) = zero(model.grid)
+
+open_boundary_mass_flux(model, bc, side, velocity) = zero(model.grid)
+
+function open_boundary_mass_inflow(model)
+    update_open_boundary_mass_fluxes!(model)
+
+    u, v, w = model.velocities
+    u_bcs = u.boundary_conditions
+    v_bcs = v.boundary_conditions
+    w_bcs = w.boundary_conditions
+
+    total_flux = zero(model.grid)
+
+    # Add flux through left boundaries
+    total_flux += open_boundary_mass_flux(model, u_bcs.west, Val(:west), u)
+    total_flux += open_boundary_mass_flux(model, v_bcs.south, Val(:south), v)
+    total_flux += open_boundary_mass_flux(model, w_bcs.bottom, Val(:bottom), w)
+
+    # Subtract flux through right boundaries
+    total_flux -= open_boundary_mass_flux(model, u_bcs.east, Val(:east), u)
+    total_flux -= open_boundary_mass_flux(model, v_bcs.north, Val(:north), v)
+    total_flux -= open_boundary_mass_flux(model, w_bcs.top, Val(:top), w)
+
+    return total_flux
+end
+
+correct_left_boundary_mass_flux!(u, bc::OBC, ::Val{:west},   avg_mass_inflow) = interior(u, 1, :, :) .-= avg_mass_inflow
+correct_left_boundary_mass_flux!(v, bc::OBC, ::Val{:south},  avg_mass_inflow) = interior(v, :, 1, :) .-= avg_mass_inflow
+correct_left_boundary_mass_flux!(w, bc::OBC, ::Val{:bottom}, avg_mass_inflow) = interior(w, :, :, 1) .-= avg_mass_inflow
+correct_left_boundary_mass_flux!(velocity, bc::IOBC, ::Val{:west}, avg_mass_inflow) = nothing
+correct_left_boundary_mass_flux!(velocity, bc::IOBC, ::Val{:south}, avg_mass_inflow) = nothing
+correct_left_boundary_mass_flux!(velocity, bc::IOBC, ::Val{:bottom}, avg_mass_inflow) = nothing
+correct_left_boundary_mass_flux!(velocity, bc, side, avg_mass_inflow) = nothing
+
+correct_right_boundary_mass_flux!(u, bc::OBC, ::Val{:east},  avg_mass_inflow) = interior(u, u.grid.Nx + 1, :, :) .+= avg_mass_inflow
+correct_right_boundary_mass_flux!(v, bc::OBC, ::Val{:north}, avg_mass_inflow) = interior(v, :, v.grid.Ny + 1, :) .+= avg_mass_inflow
+correct_right_boundary_mass_flux!(w, bc::OBC, ::Val{:top},   avg_mass_inflow) = interior(w, :, :, w.grid.Nz + 1) .+= avg_mass_inflow
+correct_right_boundary_mass_flux!(velocity, bc::IOBC, ::Val{:east}, avg_mass_inflow) = nothing
+correct_right_boundary_mass_flux!(velocity, bc::IOBC, ::Val{:north}, avg_mass_inflow) = nothing
+correct_right_boundary_mass_flux!(velocity, bc::IOBC, ::Val{:top}, avg_mass_inflow) = nothing
+correct_right_boundary_mass_flux!(velocity, bc, side, avg_mass_inflow) = nothing
+
+"""
+enforce_open_boundary_mass_conservation!(model::NonhydrostaticModel)
+
+Correct boundary mass fluxes for perturbation advection boundary conditions to ensure
+zero net mass flux through each boundary.
+"""
+function enforce_open_boundary_mass_conservation!(model)
+    u, v, w = model.velocities
+    grid = model.grid
+
+    total_mass_inflow = open_boundary_mass_inflow(model)
```

Comment:
I think things like this read better if you pair a verbose name with a symbol, so here's an idea:

```julia
∮u_dA = open_boundary_mass_inflow(model)
A = model.boundary_mass_fluxes.total_area_matching_scheme_boundaries
 A⁻¹_∮u dA = ∮u_dA / A
```

or something like that. 

because otherwise we have two names